### PR TITLE
Add authentication with Access Token

### PR DESCRIPTION
**Pull request by StefanBogdan**
fixes Issue #133

### CODE REVIEW

File: weaviate/auth.py [hunk 1]
```diff
@@ -33,12 +33,12 @@ def __init__(self, client_secret: str):
...
         ----------
         client_secret : str
-            The access token.
+            The client secret credentials, NOT access token.
         """
 
         super().__init__()
         self._credentials_body["grant_type"] = "client_credentials"
-        self.client_secret_encoded = base64.b64encode(client_secret.encode('ascii')).decode('ascii')
+        self._client_secret_encoded = base64.b64encode(client_secret.encode('utf-8')).decode('utf-8')
```
Comment (dirkkul) [inline weaviate/auth.py:41]: Can you add a test that needs utf-8 encoding?

File: test/connection/test_connection.py [hunk 2]
```diff
@@ -147,6 +147,23 @@ def test__init__(self, mock_requests, mock_refresh_authentication):
...
 
+        # 200 status_code return and token provided
+        with patch("weaviate.connect.connection.Connection._log_in") as mock_login:
+            connection = Connection(
+                url='test_url',
+                auth_client_secret=None,
+                timeout_config=(2, 20),
+                proxies=None,
+                trust_env=False,
+                additional_headers={'AuthorizatioN': 'my token'},
```
Comment (dirkkul) [inline test/connection/test_connection.py:158]: is the N at the end of AuthorizatioN a typo?

**Review by dirkkul — APPROVED** [review]

File: test/connection/test_connection.py (hunk 2 above)
Comment (StefanBogdan) [reply test/connection/test_connection.py:158]: No, header key should be case insensitive.

File: weaviate/auth.py (hunk 1 above)
Comment (StefanBogdan) [reply weaviate/auth.py:41]: Sure, I will add one.

**Comment by dirkkul** [comment]
Would you add an example to the docs on how to do this?
https://weaviate.io/developers/weaviate/current/client-libraries/python.html#authentication

File: weaviate/auth.py (hunk 1 above)
Comment (StefanBogdan) [reply weaviate/auth.py:41]: done

**Pull request merged**